### ISSUE TITLE
[BUG] class detection on WebComponent ? 

### ISSUE THREAD

**Issue by neviaumi** [francoismassart/eslint-plugin-tailwindcss#374]
**Describe the bug**
[tailwindcss/no-custom-classname
](https://github.com/neviaumi/demo-eslint-tailwindcss-bug/blob/9fdaf36aae3a8dd41e1dd890d1c4247c3851f4c7/src/awesome-timer.element.js#L15)
I am working custom element in plain JS project, and when i assign style to the html tag though `class` attributes, seem the checking was disabled. 

**To Reproduce**
1. Go the repo provided
2. run npm install && npx eslint .

**Expected behavior**
`tailwindcss/no-custom-classname` should be trigger

**Screenshots**
```js    
connectedCallback() {
        const template = document.createElement('template');
        template.innerHTML = `
<link rel="stylesheet" href="${mainCss}">
<main class="page">
    <section class="${clsx("xyz")}">
    Use clsx wrapper the class naem to trigger eslint
</section>
    <section class="page-content">
       Useing wrong class names without error
    </section>
</main>`;
        this.shadowRoot.appendChild(template.content.cloneNode(true));
    }
```
Only the class wrapped in clsx has been checked.

**Environment (please complete the following information):**
 - OS: Mac
 - Softwares + version used:
     -  Node 20
    - eslint-plugin-tailwindcss 3.17.5
    - eslint 9.13.0

**Additional context**

**eslint config file or live demo**
By providing a link to a live demo, a demo video or a github repo fixing the issue will be easier.

**Comment by neviaumi** [comment]
I am happy to warp my class with clsx as temp solution, but seem it was unexpected behaviour for me
